"-Woctal-literal" does not work in 2.13.2

In Scala 2.13.2 the `-Woctal-literal` is giving false positives. This did not happen in 2.13.1.
This issue affects `integer` and `hex` literals.

## reproduction steps
Minimal program: 
```scala
val a = 0    // Warn
val b = 0xAF // Ok
val c = 0x0F // Warn
```
Compile using:
```
scalac -deprecation -Woctal-literal Main.scala
```

## problem

```scala
Main.scala:4: warning: Decimal integer literals should not have a leading zero. (Octal syntax is obsolete.)
  val a = 0    // Warn
          ^
Main.scala:6: warning: Decimal integer literals should not have a leading zero. (Octal syntax is obsolete.)
  val c = 0x0F // Warn
          ^
2 warnings
```

## expectation
The integer value `0` should not be considered as "having a leading zero".
IMO hex litterals should still allow leading zeros as writing `0x1A 0x07` is more natural and widespread than `0x1A 0x7`.

--- a/src/compiler/scala/tools/nsc/ast/parser/Scanners.scala
+++ b/src/compiler/scala/tools/nsc/ast/parser/Scanners.scala
@@ -1150,7 +1150,7 @@ trait Scanners extends ScannersCommon {
           syntaxError("missing integer number") // e.g., 0x; previous error shadows this one
           0L
         } else {
-          if (settings.warnOctalLiteral && strVal.charAt(0) == '0')
+          if (settings.warnOctalLiteral && base == 10 && strVal.charAt(0) == '0' && strVal.length() > 1)
             deprecationWarning("Decimal integer literals should not have a leading zero. (Octal syntax is obsolete.)", since="2.10")
           convertIt
         }

--- a/test/files/neg/literals.scala
+++ b/test/files/neg/literals.scala
@@ -1,4 +1,4 @@
-// scalac: -Ywarn-octal-literal -Xfatal-warnings -deprecation
+// scalac: -Woctal-literal -Werror -deprecation
 trait RejectedLiterals {
 
   def missingHex: Int    = { 0x }        // line 4: was: not reported, taken as zero
@@ -50,4 +50,9 @@ trait Lengthy {
   def bad = 1l
 
   def worse = 123l
+}
+
+trait Regressions {
+  def oopsy = 0         // nowarn
+  def hexed = 0x0F      // nowarn
 }

